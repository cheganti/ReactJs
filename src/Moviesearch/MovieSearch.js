import React, { Component } from 'react';
import jsonData from '../data/jsonData.json';
import { MovieList } from '../Movielist/MovieList';

export default class MovieSearch extends Component {
    constructor(props) {
        super(props);
        this.jsonData = jsonData;

        this.state = {
            searchResults: this.jsonData.data,
            updatedList: this.jsonData.data,
            inputValues: ""
        };
    }
    sortByGengres = () => {
        const sortByGengresTag = this.state.searchResults
        let sortResult = sortByGengresTag.sort((a, b) => {
            if (a.genres > b.genres) {
                return 1
            } else if (a.genres < b.genres) {
                return -1
            }
        })
        this.setState({
            searchResults: sortResult
        })
    }
    sortByTitles = () => {
        const sortByTitleNames = this.state.searchResults
        let sortResult = sortByTitleNames.sort((a, b) => {
            if (a.title > b.title) {
                return 1
            } else if (a.title < b.title) {
                return -1
            }
        })
        this.setState({
            searchResults: sortResult
        })
    }

    sortByRelease = () => {
        const sortByReleaseDate = this.state.searchResults
        let sortResult = sortByReleaseDate.sort((a, b) => {
            return new Date(b.release_date) - new Date(a.release_date);
        })
        this.setState({
            searchResults: sortResult
        })
    }
    sortByRating = () => {
        const sortByMovieRating = this.state.searchResults
        let sortResult = sortByMovieRating.sort((a, b) => {
            return b.vote_count - a.vote_count
        })
        this.setState({
            searchResults: sortResult
        })
    }

    handleMovieSearch = (e) => {
        let targetValues = "";
        if (e.target.value !== '') {
            targetValues = e.target.value;
        }
        this.setState({
            inputValues: targetValues
        })
    }
    onClickResults = () => {
        const currentLists = this.state.updatedList;
        let filterValue = this.state.inputValues;
        let movieList = "";
        if (filterValue !== '') {
            movieList = currentLists.filter(item => {
                filterValue = filterValue.toLowerCase();
                return item.title.toLowerCase().includes(filterValue);
            })

        } else {
            movieList = this.jsonData.data;
        }
        this.setState({
            searchResults: movieList
        });
    }
    render() {
        const data = this.state.searchResults;
        return (
            <div>
                <div className="jumbotron">
                    <h1 className="title">Find your movie</h1>
                    <div className="input-group mb-3">
                        <input className="form-control"
                            type="text"
                            id="search"
                            placeholder="Search..." onChange={this.handleMovieSearch}
                        />
                        <div className="input-group-append">
                            <button onClick={this.onClickResults} className="btn btn-success-cus">Search</button>
                            <button onClick={this.sortByGengres} className="btn btn-success-cus">Genres</button>
                            <button onClick={this.sortByTitles} className="btn btn-success-cus">Title</button>
                        </div>
                    </div>
                    <div className="sortByDiv"><span className="bold">Sort By:</span>
                        <button onClick={this.sortByRelease} className="btn btn-success-cus">Release Date</button>
                        <button onClick={this.sortByRating} className="btn btn-success-cus">Rating</button>
                    </div>
                </div>
                <MovieList data={data} sortByTitles={this.sortByTitles} />
            </div>
        )
    }
}