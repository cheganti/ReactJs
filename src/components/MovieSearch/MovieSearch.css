h1.title {
  color: #ffffff;
  text-transform: uppercase;
  font-weight: normal;
  position: absolute;
  top: 7%;
}
