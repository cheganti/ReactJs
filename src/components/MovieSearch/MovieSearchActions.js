
export const UPDATE_SEARCH_TEXT = 'UPDATE_SEARCH_TEXT';

export const updateSearchText = searchText => ({ type: UPDATE_SEARCH_TEXT, searchText });
