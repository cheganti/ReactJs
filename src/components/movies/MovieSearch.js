import React, { Component } from 'react';
import jsonData from '../data/jsonData';

export class MovieSearch extends Component {
    constructor(props) {
        super(props);
        this.jsonData = jsonData;
        this.state = {
            searchValues: this.jsonData.data,
        };
    }
    handleChange = (e) => {
        console.log(e)
        console.log(this.state.searchValues)
        this.setState({
            // searchValues: newList
        });
    }
    render() {
        return (
            <div className="search jumbotron">
                <input
                    type="text"
                    id="search"
                    placeholder="Search..." onChange={this.handleChange}
                />
                <button>Search</button>
                <button>Genres</button>
                <button>Title</button>
            </div>
        )
    }
}
