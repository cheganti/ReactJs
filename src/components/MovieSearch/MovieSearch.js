import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import './MovieSearch.css';
import { HandleInputs } from '../../store/actions';

function handleMovieSearch(e) {
  let targetValues = '';
  if (e.target.value !== '') {
    targetValues = e.target.value;
  }
  const { handleInputs } = this.props;
  handleInputs(targetValues);
}

function MovieSearch(props) {
  const {
    onClickResults,
    searchByTitles,
    isTitle,
    searchByGengres,
    isGenres,
  } = props;
  return (
    <div>
      <h1 className="title">Find your movie</h1>
      <div className="input-group mb-3">
        <input
          className="form-control"
          type="text"
          id="search"
          placeholder="Search..."
          onChange={handleMovieSearch}
        />
        <div className="input-group-append">
          <button
            type="button"
            onClick={onClickResults}
            className="btn btn-success-cus"
          >
            Search
          </button>
          <button
            type="button"
            onClick={searchByTitles}
            className={`btn btn-success-cus ${isTitle === true
              ? 'active'
              : 'deactive'}`}
          >
            Title
          </button>
          <button
            type="button"
            onClick={searchByGengres}
            className={`btn btn-success-cus ${isGenres === true
              ? 'active'
              : 'deactive'}`}
          >
            Genres
          </button>
        </div>
      </div>
    </div>
  );
}
MovieSearch.propTypes = {
  isTitle: PropTypes.bool.isRequired,
  isGenres: PropTypes.bool.isRequired,
  handleMovieSearch: PropTypes.func.isRequired,
  onClickResults: PropTypes.func.isRequired,
  searchByTitles: PropTypes.func.isRequired,
  searchByGengres: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  results: state.results,
  isTitle: state.isTitle,
  isGenres: state.isGenres,
  isRelease: state.isRelease,
  isRating: state.isRating,

});

const mapDispatchToProps = dispatch => ({
  handleInputs: (data) => {
    dispatch(HandleInputs(data));
  },
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(MovieSearch);
