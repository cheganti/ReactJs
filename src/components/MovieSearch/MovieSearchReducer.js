import { UPDATE_SEARCH_TEXT } from './MovieSearchActions';

function MovieSearchReducer(state = {
  inputs: '',
}, action) {
  switch (action.type) {
    case UPDATE_SEARCH_TEXT:
      return {
        ...state,
        inputs: action.searchText,
      };

    default:
      return state;
  }
}
export default MovieSearchReducer;
