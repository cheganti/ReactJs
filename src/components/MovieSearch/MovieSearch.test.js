import React from 'react';
import { shallow } from 'enzyme';
import MovieSearch from './MovieSearch';


describe('MovieSorting', () => {
  it('renders without breaking', () => {
    const OnChangeSearch = jest.fn();
    const onClickSearch = jest.fn();
    const handleSearchByTitles = jest.fn();
    const handleSearchByGenres = jest.fn();
    const wrapper = shallow(<MovieSearch handleMovieSearch={OnChangeSearch} onClickResults={onClickSearch} searchByTitles={handleSearchByTitles} searchByGengres={handleSearchByGenres} />);
    expect(wrapper).toMatchSnapshot();
  });
});
