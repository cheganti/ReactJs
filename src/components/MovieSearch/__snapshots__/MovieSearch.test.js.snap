// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieSorting renders without breaking 1`] = `
<div>
  <h1
    className="title"
  >
    Find your movie
  </h1>
  <div
    className="input-group mb-3"
  >
    <input
      className="form-control"
      id="search"
      onChange={[MockFunction]}
      placeholder="Search..."
      type="text"
    />
    <div
      className="input-group-append"
    >
      <button
        className="btn btn-success-cus"
        onClick={[MockFunction]}
        type="button"
      >
        Search
      </button>
      <button
        className="btn btn-success-cus deactive"
        onClick={[MockFunction]}
        type="button"
      >
        Title
      </button>
      <button
        className="btn btn-success-cus deactive"
        onClick={[MockFunction]}
        type="button"
      >
        Genres
      </button>
    </div>
  </div>
</div>
`;
